### DISCUSSION TITLE
pdf preview lf & kitty

### DISCUSSION THREAD

**Discussion by kristoferus75** [gokcehan/lf#1207] [Q&A] [answered]
Hi !

I already use lf filemanager with kitty to see images preview !

but pdf preview doesnt work

Please help !

Thanks !

heres my preview script:

```
#!/usr/bin/env bash

file=$1

w=$2

h=$3

x=$4

y=$5


CACHE="$HOME/.cache/tmpimg"


filetype="$( file -Lb --mime-type "$file")"

if [[ "$filetype" =~ ^image ]]; then

kitty +kitten icat --silent --stdin no --transfer-mode file --place "${w}x${h}@${x}x${y}" "$file" < /dev/null > /dev/tty

exit 1

fi

if [[ "$filetype" =~ pdf/* ]]; then

pdftoppm -jpeg -f 1 -singlefile "$file" "$CACHE"

kitty +kitten icat --silent --transfer-mode file --place "${w}x${h}@${x}x${y}" "${CACHE}.jpg"

exit 1

fi
```

**Comment by joelim-work** [comment]
You script appears to just simply invoke `pdftoppm` to generate a PDF preview image and then display it in the terminal, assuming the file type is detected properly. If that is the case, then you should be able to just run the script directly (i.e. without `lf`), and just provide the filename and some dummy coordinates:

```shell
./my_preview_script /path/to/pdf/file 1 2 3 4
```

Ensure that the image is generated properly and then displayed in the terminal. You can also add `set -x` to your script to print debugging messages.

---

BTW the conditional expression `"$filetype" =~ pdf/*` looks a bit strange to me. The operator `=~` means to match using a regex, so it will match `pdf` followed by any number of `/`. It's probably better if you just match the file type directly like `"$filetype" = application/pdf`.

**Reply by kristoferus75** [reply]
Hi !

Thanks  i have adapt the script and there is no error !

but the preview picture is extreme small (only a pixel)

sh my_preview_script.sh ~/test.pdf 1 2 3 4
+ CACHE=/home/user/.cache/tmpimg
++ file -Lb --mime-type /home/user/test.pdf
+ filetype=application/pdf
+ [[ application/pdf =~ ^image ]]
+ [[ application/pdf = application/pdf ]]
+ pdftoppm -jpeg -f 1 -singlefile /home/user/test.pdf /home/user/.cache/tmpimg
+ kitty +kitten icat --silent --transfer-mode file --place 1x2@3x4 /home/user/.cache/tmpimg.jpg

If i open the image from the .cache folder (tmpimg)  it looks ok

**Reply by joelim-work** [reply]
If the preview picture is too small, then try bigger numbers, I just used 1,2,3,4 as an example. I don't use Kitty myself so I'm not sure what values are appropriate.

You can also try adding the line `echo "$@" >> ~/lf.log` to your script. This will write the script arguments to a log file so you can see what values are being passed in when running `lf`.

**Reply by kristoferus75** [reply]
o thanks thats cool with the log !  I don't know why but it work now !  Here is my finish adapt pdf part:

```

if [[ "$filetype" = application/pdf ]]; then
    pdftoppm -jpeg -f 1 -singlefile "$file" "$CACHE"
    kitty +kitten icat --silent --stdin no --transfer-mode file --place "${w}x${h}@${x}x${y}" "${CACHE}.jpg" < /dev/null > /dev/tty
    exit 1
fi

```

**Reply by joelim-work** [reply, marked as answer]
If it's working for you now then please mark this discussion as answered. But anyway I'm glad you managed to solve your issue 🙂

**Reply by kristoferus75** [reply]
Thanks alot for your help !!!!
